### PR TITLE
Pass previous rendered block type, and render a hr on adjacent text blocks

### LINKED ISSUE #560: Investigate how rendered blocks could know context
Blocks need to know context for their render, for example where they render relative to other blocks, what blocks have rendered above/below them, which they currently do not know.

We should investigate how we would add that context.

An example use-case for this is https://github.com/bbc/simorgh/issues/452.

### PR DESCRIPTION
**Pull request by jtart**
Resolves #560

_Abstracts the type of the previous rendered block and passes it to the component of a block. Then renders a horizontal rule if the previous block type matches `text`._

- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: src/app/containers/MainContent/index.jsx [hunk 1]
```diff
@@ -19,12 +19,21 @@ const Blocks = {
 };
 
 const render = blocks =>
-  blocks.map(block => {
+  blocks.map((block, index) => {
     const { type, blockId, model } = block;
 
+    const { type: typeOfPreviousBlock } = blocks[index - 1] || {};
```
Comment (pjlee11) [inline src/app/containers/MainContent/index.jsx:25]: To clarify this only addresses top level blocks and their previous siblings. Therefore this tells the block `n` what it's older sibling is `n-1`. Also due to it being embedded in the map of MainContent it only handles top level blocks and doesn't affect any nested blocks.

Comment (jtart) [reply src/app/containers/MainContent/index.jsx:25]: Yes it only handles top level blocks. We currently don't map through nested blocks in the same way so it doesn't need to be applied elsewhere. Easy to make the switch though.

File: src/app/containers/Text/index.jsx [hunk 2]
```diff
@@ -1,13 +1,32 @@
...
+import { string } from 'prop-types';
 import { textModelPropTypes } from '../../models/propTypes/text';
 import Text from '../../components/Text';
 
-const TextContainer = ({ blocks }) => {
+const TextContainer = ({ type, blocks, typeOfPreviousBlock }) => {
   if (!blocks) return null;
 
-  return blocks.map(({ blockId, model }) => <Text key={blockId} {...model} />);
+  const HorizontalRule = typeOfPreviousBlock === type ? <hr /> : null;
```
Comment (sareh) [inline src/app/containers/Text/index.jsx]: I'd suggest extracting this check into a function, for readability. `shouldPrependHR` or something of the like.

**Review by pjlee11 — DISMISSED** [review]
LGTM nice work

**Review by sareh — COMMENTED** [review]
Looks great. Just a comment about refactoring.

**Comment by sareh** [comment]
When checking locally at this test article, http://localhost:7080/news/articles/c0000000011o 
I see the following - text blocks separated by horizontal rules. 👍 
screen shot article 11

**Review by sareh — DISMISSED** [review]
👍 Looks great!

**Review by pjlee11 — DISMISSED** [review]
Nice work, thanks for making tests more descriptive and adding the `shouldPrependHR` func 👍

**Pull request merged**